crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -423,42 +423,29 @@ impl<'w, 's> Commands<'w, 's> {
     /// #[derive(Component)]
     /// struct Agility(u32);
     ///
-    /// fn example_system(mut commands: Commands) {
+    /// fn example_system(mut commands: Commands) -> Result {
     ///     // Create a new, empty entity
     ///     let entity = commands.spawn_empty().id();
     ///
-    ///     commands.entity(entity)
+    ///     commands.entity(entity)?
     ///         // adds a new component bundle to the entity
     ///         .insert((Strength(1), Agility(2)))
     ///         // adds a single component to the entity
     ///         .insert(Label("hello world"));
+    ///     Ok(())
     /// }
     /// # bevy_ecs::system::assert_is_system(example_system);
     /// ```
```

Comment:
I'm confused by the objective of this PR. To me `commands.entity(entity)` is like indexing a slice when you know the index is valid. e.g.

```rust
for entity in query.iter() {
  commands.entity(entity)
}
```

> This will be high impact and pushes users heavily towards the `?` pattern, but I think it's the right call. Runtime panics are a serious hazard in production Bevy code, and these sorts of very hard to predict panics when working with commands are extremely frustrating.

Many of the changes here simply do an `unwrap`. The other changes to early return with `?` make it harder to find where a panic *should* have happened when the entity should have already existed.